docs/client-sdk-states-and-lifecycle.mdx

```diff
@@ -0,0 +1,414 @@
+
+
+## Server-Side Experiment States
+
+The Nimbus experiment publishing workflow moves each experiment through several different states, reflected in the experiment data published to the Remote Settings server. Transitions between the states are triggered by updates published to Remote Settings. (TODO: maybe also by the passage of time as observed by the RS server?)
+
+Note that we are only interested in states that are observable by the client; the experiment publishing workflow has additional states for internal use that are not represented here.
+
+
+
+![drawing](/img/client-sdk-states-and-lifecycle/client-visible-publishing-workflow-states.png)
+
+
+#### The "Preparing" State
+
+Before an experiment becomes visible to clients in Remote Settings, it may go through a series of drafts and adjustments and approvals in the Experimenter console. This is represented in the server-side workflow by several different states, but since it is completely invisible to the Client SDK these will not be discussed further in this document.
+
+Transitions:
+
+
+
+*   To **Enrolling **state, by being made visible to clients in Remote Settings.
+
+
+#### The "Enrolling" State
+
+This is the first phase of an experiment going "live" and starting to show up in the user experience. It's indicated by the experiment being visible to clients in Remote Settings with _isEnrollmentPaused_ set to _false_.
+
+Clients that _observe an experiment transition_ to this state should check whether to enroll in the experiment (N.B. this includes a self-transition from **Enrolling** -> **Enrolling** that is accompanied by a change in the published experiment config, which might change enrollment decisions).
+
+While the experiment is in this state, enrolled clients should activate the experiment treatment appropriate for their enrolled branch, and report telemetry about it.
+
+Transitions:
+
+
+
+*   To **Running** state, when a Remote Settings update changes _isEnrollmentPaused_ to _true_.
+*   To **Enrolling **state, when a Remote Settings update changes other experiment parameters.
+*   To **Ended** state, if the experiment ceases to be visible in Remote Settings.
+
+
+#### The "Running" State
+
+This is the second phase of an experiment being live and showing up in the user experience. It's indicated by the experiment being visible to clients in Remote Settings with _isEnrollmentPaused_ set to _true_.
+
+While the experiment is in this state, clients that are _not_ already enrolled in the experiment should not enroll themselves, even if the experiment config changes in a way that would otherwise cause them to enroll. Already-enrolled clients should continue to activate the experiment treatment appropriate for their enrolled branch, and report telemetry about it.
+
+Transitions:
+
+
+
+*   To **Enrolling **state, when a Remote Settings update changes _isEnrollmentPaused_ to _false_.
+*   To **Running **state, when a Remote Settings update changes other experiment parameters.
```

Comment:
Same as above, I _think_ this might need to be `**Enrolling** state,` and `**Running** state,` above to ensure boldedness.

Reply:
Thanks again; will fix here too.